src/azul/indexer/__init__.py

```diff
@@ -192,6 +192,15 @@ class SourceSpec(ABC, Generic[SOURCE_SPEC]):
     def parse(cls, spec: str) -> SOURCE_SPEC:
         raise NotImplementedError
 
+    @classmethod
+    def _parse(cls, spec: str) -> Tuple[str, Prefix]:
+        rest, sep, prefix = spec.rpartition(':')
+        reject(sep == '',
```

Comment:
Consider a DSS source spec which always has a colon after the scheme. This wouldn't get hit and something else would break much further down the line. For a TDR source spec it would fail during parsing of the rest. In either case the reject wouldn't kick in and a confusing error would be raised. I don't think we should pretend that this catches the absence of the prefix segment and have a more general error message. Or we pick a different separator, one that we know doesn't occur in the rest.

Reply:
Opted to change the error message to something more generic. #3494 will remove the need for the separators.